gateway/src/tls.rs

```diff
@@ -7,16 +7,40 @@ use std::sync::Arc;
 use axum_server::accept::DefaultAcceptor;
 use axum_server::tls_rustls::{RustlsAcceptor, RustlsConfig};
 use futures::executor::block_on;
+use http::StatusCode;
 use pem::Pem;
 use rustls::server::{ClientHello, ResolvesServerCert};
 use rustls::sign::{self, CertifiedKey};
 use rustls::{Certificate, PrivateKey, ServerConfig};
 use rustls_pemfile::Item;
-use shuttle_common::models::error::ErrorKind;
+use shuttle_common::models::error::ApiError;
+use thiserror::Error;
 use tokio::runtime::Handle;
 use tokio::sync::RwLock;
 
-use crate::Error;
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Tls(#[from] rustls::Error),
+
+    #[error("Got an unknown certificate")]
+    UnknownType,
+
+    #[error("read failed: {0}")]
+    Io(#[from] std::io::Error),
+
+    #[error("failed to get signing key: {0}")]
+    SigningKey(#[from] sign::SignError),
+}
+
+impl From<Error> for ApiError {
+    fn from(err: Error) -> Self {
+        Self {
+            message: err.to_string(),
+            status_code: StatusCode::INTERNAL_SERVER_ERROR.as_u16(),
+        }
+    }
+}
```

Comment:
I believe only we see these errors when setting up custom domains - so wanted to return as much as possible when these do fail.

What do you mean by "error event"?

Reply:
Ah, in that case it does make sense. Perhaps we should add a comment about that? 

By error event I mean tracing::error events, but I don't think it will be very useful if this will just be called with the admin client.

Reply:
I'm just confirming where we see these errors...

Reply:
This only appears in the create custom domain and renew custom domain so we wand it to be verbose